Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,54 @@
+
+# Managing Assets
+
+* Author(s): Adrian Śliwa
+* Review Manager: TBD
+
+## Introduction
+
+Babylon's iOS application is a large and advanced software project. This project consisting of many frameworks. Each of them consists of many tools or features, most of which need icons or images to be fully implemented. The goal of this proposal is to standardise the way of adding, changing, removing and accessing assets in our codebase.
+
+## Motivation
+
+Managing assets is not difficult engineering problem but without standardised rules we can encounter following issues:
+1. It can be difficult to find if a given icon or image is already included in one of the frameworks. At the moment we have icons and images in many places.
+2. It can cause duplications of icons if the existing icon will be added again.
+3. It could be impossible to reuse icon or image included in one of the frameworks which is not linked to the framework where someone is implementing a feature.
+4. We can end up with many duplicate solutions for accessing icons or images in the code.
+
+To make sure we won't have duplicated icons in the project and all icons will be accessible in all places where we implement the user interface, all of us should agree on uniform rules how and where we should add new assets and how we should manage them.
+
+
+## Proposed solution
+
+The proposed solution contains a set of rules which should be followed during manipulation of our assets:
+1. All new icons and images we are going to use should be added to `BabylonDependencies.framework` in `Assets.xcassets` catalog 
+2. `struct DesignLibrary.Tokens` should have another property for icons `public let icons: Icons`
+3. Icons should be namespaced and put into `struct` with common relation/meaning (e.g. `TabBarIcons`, `Iconography` etc.). If there is no existing `struct` we should create one. Then we have to add a new case to `enum ImageIdentifier` which is embedded in given `struct`.
+4. To access e.g. iconography's close icon we could use subscripts `designLibrary.tokens.icons.iconography[.close]`
```

Comment:
this seems overly verbose, I'd suggest we have it reduced to `designLibrary.icons[.close]` and have all these details inside this subscript if we need them for some reason

Reply:
Which is how it's working now, despite images being defined in `Iconography`.